kie-server-parent/kie-server-services/kie-server-services-prometheus/src/main/java/org/kie/server/services/prometheus/PrometheusKieServerExtension.java

```diff
@@ -0,0 +1,134 @@
+
+    public static final String EXTENSION_NAME = "Prometheus";
+
+    private static final Boolean disabled = Boolean.parseBoolean(System.getProperty(KieServerConstants.KIE_PROMETHEUS_SERVER_EXT_DISABLED, "true"));
+
+    public static final CollectorRegistry registry = CollectorRegistry.defaultRegistry;
+
+    private KieServerRegistry context;
+
+    private List<Object> services = new ArrayList<>();
+    private boolean initialized = false;
+
+    private static PrometheusMetrics METRICS = null;
+
+    public static PrometheusMetrics getMetrics() {
+        if (METRICS == null) {
+            METRICS = new PrometheusMetrics();
+        }
+        return METRICS;
+    }
+
+    @Override
+    public boolean isInitialized() {
+        return initialized;
+    }
+
+    @Override
+    public boolean isActive() {
+        return !disabled;
+    }
+
+    @Override
+    public void init(KieServerImpl kieServer, KieServerRegistry registry) {
+        this.context = registry;
+        initialized = true;
+    }
+
+    @Override
+    public void destroy(KieServerImpl kieServer, KieServerRegistry registry) {
+        //no-op
+    }
+
+    @Override
+    public void createContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        //no-op
+    }
+
+    @Override
+    public boolean isUpdateContainerAllowed(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        return true;
+    }
+
+    @Override
+    public void updateContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        //no-op
+    }
+
+    @Override
+    public void disposeContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        //no-op
+    }
+
+    @Override
+    public List<Object> getAppComponents(SupportedTransports type) {
+
+        ServiceLoader<KieServerApplicationComponentsService> appComponentsServices
+                = ServiceLoader.load(KieServerApplicationComponentsService.class);
+        List<Object> appComponentsList = new ArrayList<Object>();
+        Object[] services = {context};
+        for (KieServerApplicationComponentsService appComponentsService : appComponentsServices) {
+            appComponentsList.addAll(appComponentsService.getAppComponents(EXTENSION_NAME, type, services));
+        }
+        return appComponentsList;
+    }
+
+    @Override
+    public <T> T getAppComponents(Class<T> serviceType) {
+        return null;
+    }
+
+    @Override
+    public String getImplementedCapability() {
+        return KieServerConstants.CAPABILITY_PROMETHEUS;
+    }
+
+    @Override
+    public List<Object> getServices() {
+        return services;
+    }
+
+    @Override
+    public String getExtensionName() {
+        return EXTENSION_NAME;
+    }
+
+    @Override
+    public Integer getStartOrder() {
+        return Integer.MAX_VALUE;
```

Comment:
shouldn't this be one of the first to init as others might need access to that extension?
https://github.com/kiegroup/droolsjbpm-integration/pull/1708/files#diff-dd2d9058a692175656a6a2553d7195ddR166

Reply:
You're right but I can't see why the extension is running correctly even by being the last. 

Anyway DMN has 47 and Drools 0, KieContainer has -1. 

Can I push Drools a bit further (let's say 5) and put this to 2?

We should use Basic numbers for these (10,20,30...)